cmd/snap/cmd_run.go

```diff
@@ -1103,140 +1110,304 @@ func (x *cmdRun) runCmdWithTraceExec(origCmd []string, envForExec envForExecFunc
 	}
 	defer fw.Close()
 
+	appCmd := exec.Command(straceShim, origCmd...)
+	appCmd.Stdin = os.Stdin
+	appCmd.Stdout = os.Stdout
+	appCmd.Stderr = os.Stderr
+	appCmd.Env = envForExec(nil)
+	if err := appCmd.Start(); err != nil {
+		return err
+	}
+
+	// wait for the child process executing gdb helper to raise SIGSTOP
+	// signalling readiness to attach a gdbserver process
+	var status syscall.WaitStatus
+	_, err = syscall.Wait4(appCmd.Process.Pid, &status, syscall.WSTOPPED, nil)
```

Comment:
The shim is simple so there's little risk of another signal being raised, but if something other than SIGTOP is raised then we'll just hang here, correct? We could wait for any signal and then check `status.Stopped()`. But there isn't much risk